tests/public.rs

```diff
@@ -0,0 +1,101 @@
+                    assert_eq!(track.media_time, Some(mp4::TrackScaledTime(0, 0)));
+                    assert_eq!(track.timescale, Some(mp4::TrackTimeScale(12800, 0)));
+                    assert_eq!(v.width, 320);
+                    assert_eq!(v.height, 240);
+
+                    // track.tkhd part
+                    let tkhd = track.tkhd.unwrap();
+                    assert_eq!(tkhd.disabled, false);
+                    assert_eq!(tkhd.duration, 40);
+                    assert_eq!(tkhd.width, 20971520);
+                    assert_eq!(tkhd.height, 15728640);
+
+                    // track.data part
+                    assert_eq!(match v.codec_specific {
+                        mp4::VideoCodecSpecific::AVCConfig(v) => {
+                            assert!(v.len() > 0);
+                            "AVC"
+                        }
+                        mp4::VideoCodecSpecific::VPxConfig(vpx) => {
+                            // We don't enter in here, we just check if fields are public.
+                            assert!(vpx.bit_depth > 0);
+                            assert!(vpx.color_space > 0);
+                            assert!(vpx.chroma_subsampling > 0);
+                            assert!(vpx.codec_init.len() > 0);
+                            "VPx"
+                        }
+                    }, "AVC");
+                }
+                Some(mp4::SampleEntry::Audio(a)) => {
+                    // track part
+                    assert_eq!(track.duration, Some(mp4::TrackScaledTime(2944, 1)));
+                    assert_eq!(track.empty_duration, Some(mp4::MediaScaledTime(0)));
+                    assert_eq!(track.media_time, Some(mp4::TrackScaledTime(1024, 1)));
+                    assert_eq!(track.timescale, Some(mp4::TrackTimeScale(48000, 1)));
+
+                    // track.tkhd part
+                    let tkhd = track.tkhd.unwrap();
+                    assert_eq!(tkhd.disabled, false);
+                    assert_eq!(tkhd.duration, 62);
+                    assert_eq!(tkhd.width, 0);
+                    assert_eq!(tkhd.height, 0);
+
+                    // track.data part
+                    assert_eq!(match a.codec_specific {
+                        mp4::AudioCodecSpecific::ES_Descriptor(v) => {
+                            assert!(v.len() > 0);
+                            "ES"
+                        }
+                        mp4::AudioCodecSpecific::OpusSpecificBox(opus) => {
+                            // We don't enter in here, we just check if fields are public.
+                            assert!(opus.version > 0);
+                            "Opus"
+                        }
+                    }, "ES");
+                    assert!(a.samplesize > 0);
+                    assert!(a.samplerate > 0);
+                }
+                Some(mp4::SampleEntry::Unknown) | None => {}
+            }
+        }
+    } else {
+        panic!("read_mp4 function failed")
```

Comment:
Same here, but more so. `if let Ok(())` is cute but the `else` clause is far removed from its conditional. Early returns are better.